config/routes.rb

```diff
@@ -11,4 +11,14 @@
       resources :likes, only: [:create]
     end
   end
+
+  namespace :api, defaults: { format: :json } do
+    namespace :v1 do
+      resources :users, only: %i[index show ] do
+        resources :posts, only: %i[index] do
+          resources :comments, only: %i[index create]
+        end
+      end
+    end
+  end

```

Comment:
- [ ] In addition as the requirements says you need to add an API endpoint for authentication, both registration and login
* Login endpoint - **POST :** `/log in`
* Registration endpoint - **POST :** `/register` or `/signup`